realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -57,12 +74,110 @@ void RealSenseNodeFactory::getDevice(rs2::device_list list)
 		else
 		{
 			bool found = false;
+
+			if (_accel_orientation.empty()) // generate accel orientation vector from camera quaternion
+			{
+				try {
+					geometry_msgs::TransformStamped transformStamped = 
+									_tf_buffer.lookupTransform(_tf_link_name, "base_link", ros::Time(0));
+
+					tf2::Quaternion q_orig, q_rot;
+					// universal gravity vector
+					q_orig.setValue(0.0, -9.81, 0.0);
+					tf2::convert(transformStamped.transform.rotation , q_rot);
+					// Calculate the new orientation
+					q_orig = q_rot.inverse() * q_orig * q_rot; 
```

Comment:
I would prefer if this was cleaned, such that it is more readable and the following questions resolve themselves:
- what is setValue with 3 arguments for a quaternion supposed to be?
- why is the gravity in the y entry?
- where did you get the q = Q^-1 * q * Q formula from?

The way I would clean this up would be like this (not sure if this will compile exactly like this):
```
tf2::Matrix3x3 rot_link_base;
tf2::convert(transformStamped.transform.rotation , rot_link_base);
const tf2::Vector3 base_g(0,0,9.81);
const tf2::Vector3 link_g = rot_link_base * base_g;
```

Reply:
This is link to the formula: https://gamedev.stackexchange.com/questions/28395/rotating-vector3-by-a-quaternion

Reply:
Gravity vector is such that because in the coordinate system assumed by Intel RS is X->pointing to right, Y->pointing up.  Z->Coming out of the plane. Hence gravity (pointing downwards) shows up as 0.0, -9.81, 0.0 m/s2. We are inverse transforming this coordinate system with the relation has with base_link.

Reply:
Done. changed the code to be more clear and explainable. Added more comments to clear logic.

Still using quaternion based rotation transformation instead of 3X3 rot vec. Reason, getting some error in tf2::convert to 3X3 rot vec. It always returns in Q form. 

This code is one time execution during infrastructure boot. So overhead has no effect on robot running.